File: Core/Entities/Ticket.cs
using System.ComponentModel.DataAnnotations;
using Core.Enums;

namespace Core.Entities;

public class Ticket : BaseEntity
{
    public const int TitleMaxLength = 128;
    public const int DescriptionMaxLength = 256;
    
    [MaxLength(TitleMaxLength)]
    public required string Title { get; set; }

    [MaxLength(DescriptionMaxLength)]
    public string? Description { get; set; }

    public required TicketStatus Status { get; set; }

    public required TicketPriority Priority { get; set; }
}

File: client/src/app/shared/models/ticket.ts
export type Ticket = {
  id: number;
  createdAt?: string | null;
  updatedAt?: string | null;
  title: string;
  description?: string | null;
  status: TicketStatus;
  priority: TicketPriority;
};

export type TicketStatus = 'Open' | 'InProgress' | 'Resolved' | 'Closed';

export type TicketPriority = 'Low' | 'Medium' | 'High';
